manager/manager_test.go

```diff
@@ -164,3 +192,210 @@ func TestManager(t *testing.T) {
 	// error.
 	<-done
 }
+
+// Tests locking and unlocking the manager and key rotations
+func TestManagerLockUnlock(t *testing.T) {
+	ctx := context.Background()
+
+	temp, err := ioutil.TempFile("", "test-manager-lock")
+	require.NoError(t, err)
+	require.NoError(t, temp.Close())
+	require.NoError(t, os.Remove(temp.Name()))
+
+	defer os.RemoveAll(temp.Name())
+
+	stateDir, err := ioutil.TempDir("", "test-raft")
+	require.NoError(t, err)
+	defer os.RemoveAll(stateDir)
+
+	tc := testutils.NewTestCA(t)
+	defer tc.Stop()
+
+	managerSecurityConfig, err := tc.NewNodeConfig(ca.ManagerRole)
+	require.NoError(t, err)
+
+	_, _, err = managerSecurityConfig.KeyReader().Read()
+	require.NoError(t, err)
+
+	m, err := New(&Config{
+		RemoteAPI:      RemoteAddrs{ListenAddr: "127.0.0.1:0"},
+		ControlAPI:     temp.Name(),
+		StateDir:       stateDir,
+		SecurityConfig: managerSecurityConfig,
+		// start off without any encryption
+	})
+	require.NoError(t, err)
+	require.NotNil(t, m)
+
+	done := make(chan error)
+	defer close(done)
+	go func() {
+		done <- m.Run(ctx)
+	}()
+
+	opts := []grpc.DialOption{
+		grpc.WithTimeout(10 * time.Second),
+		grpc.WithTransportCredentials(managerSecurityConfig.ClientTLSCreds),
+	}
+
+	conn, err := grpc.Dial(m.Addr(), opts...)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, conn.Close())
+	}()
+
+	// check that there is no kek currently - we are using the API because this
+	// lets us wait until the manager is up and listening, as well
+	var cluster *api.Cluster
+	client := api.NewControlClient(conn)
+
+	require.NoError(t, raftutils.PollFuncWithTimeout(nil, func() error {
+		resp, err := client.ListClusters(ctx, &api.ListClustersRequest{})
+		if err != nil {
+			return err
+		}
+		if len(resp.Clusters) == 0 {
+			return fmt.Errorf("no clusters yet")
+		}
+		cluster = resp.Clusters[0]
+		return nil
+	}, 1*time.Second))
+
+	require.Nil(t, cluster.UnlockKeys)
+
+	// tls key is unencrypted, but there is a DEK
+	key, err := ioutil.ReadFile(tc.Paths.Node.Key)
+	require.NoError(t, err)
+	keyBlock, _ := pem.Decode(key)
+	require.NotNil(t, keyBlock)
+	require.False(t, x509.IsEncryptedPEMBlock(keyBlock))
+	require.Len(t, keyBlock.Headers, 2)
+	currentDEK, err := decodePEMHeaderValue(keyBlock.Headers[pemHeaderRaftDEK], nil)
+	require.NoError(t, err)
+	require.NotEmpty(t, currentDEK)
+
+	// update the lock key - this may fail due to update out of sequence errors, so try again
+	for i := 0; i < 3; i++ {
```

Comment:
I'm not sure - I haven't seen it fail yet.  I can also just keep going until it succeeds...